master/buildbot/reporters/bitbucketserver.py

```diff
@@ -105,44 +107,59 @@ def send(self, build):
                           code=response.code, content=content)
 
 
-class BitbucketServerPRCommentPush(http.HttpStatusPushBase):
+class BitbucketServerPRCommentPush(notifier.NotifierBase):
     name = "BitbucketServerPRCommentPush"
 
     @defer.inlineCallbacks
-    def reconfigService(self, base_url, user, password, text=None,
-                        verbose=False, **kwargs):
-        yield http.HttpStatusPushBase.reconfigService(
-            self, wantProperties=True, **kwargs)
-        self.text = text or Interpolate('Builder: %(prop:buildername)s '
-                                        'Status: %(prop:statustext)s')
+    def reconfigService(self, base_url, user, password, messageFormatter=None,
+                        verbose=False, debug=None, verify=None, **kwargs):
+        yield notifier.NotifierBase.reconfigService(
+            self, messageFormatter=messageFormatter, watchedWorkers=None,
+            messageFormatterMissingWorker=None, subject='', addLogs=False,
+            addPatch=False, **kwargs)
         self.verbose = verbose
         self._http = yield httpclientservice.HTTPClientService.getService(
             self.master, base_url, auth=(user, password),
-            debug=self.debug, verify=self.verify)
+            debug=debug, verify=verify)
 
-    @defer.inlineCallbacks
-    def send(self, build):
-        if build['complete'] and "pullrequesturl" in build['properties']:
-            yield self.sendPullRequestComment(build)
+    def checkConfig(self,base_url, user, password, messageFormatter=None,
+                    verbose=False, debug=None, verify=None, **kwargs):
+
+        notifier.NotifierBase.checkConfig(self,
+                                          messageFormatter=messageFormatter,
+                                          watchedWorkers=None,
+                                          messageFormatterMissingWorker=None,
+                                          subject='',
+                                          addLogs=False,
+                                          addPatch=False,
+                                          **kwargs)
+
+    def isMessageNeeded(self, build):
+        if 'pullrequesturl' not in build['properties']:
```

Comment:
At the moment you use the `pullrequesturl` if it's not not in the build properties. It would be neater to reverse the logic in this switch:
```
if 'pullrequesturl' in build['properties']:
  return do_something_with(url)
return False
```